gotfparse/pkg/converter/converter.go

```diff
@@ -36,118 +33,276 @@ type terraformMeta struct {
 func (t *terraformConverter) VisitJSON() *gabs.Container {
 	jsonOut := gabs.New()
 
-	var visitor func(b *terraform.Block, parentKey string)
+	for _, m := range t.modules {
+		t.visitModule(m, jsonOut)
+	}
+
+	return jsonOut
+}
+
+// visitModule takes a module and walks each of the blocks underneath it.
+func (t *terraformConverter) visitModule(m *terraform.Module, out *gabs.Container) {
+	path := t.getModulePath(m)
+
+	for _, b := range m.GetBlocks() {
+		t.visitBlock(b, path, out)
+	}
+}
+
+// visitBlock takes a block, and either builds a json model of the resource or ignores it.
+func (t *terraformConverter) visitBlock(b *terraform.Block, parentPath string, jsonOut *gabs.Container) {
+	switch b.Type() {
+	// These blocks don't have to conform to policies, and they don't have
+	//children that should have policies applied to them, so we ignore them.
+	case "data", "locals", "output", "provider", "terraform", "variable":
```

Comment:
afaics we do need all of these. data for resolution of value on usage, provider for default tags propagation, terraform for provider constraints policies and origins, output to scan for passwords, etc. locals and variables are potentially less useful, but given we're having some issues resolving locals, including both in the output is useful for debugging.

Reply:
ah, didn't know that. makes sense when you explain it though. I'll get those added back. thanks!